api/object.go

```diff
@@ -273,6 +273,44 @@ func GetObject(bp BaseParams, bck cmn.Bck, object string, options ...GetObjectIn
 	return resp.n, nil
 }
 
+// GetObjectReader returns reader of the requested object. It does not read body
+// bytes, nor validates a checksum. Caller is responsible for closing the reader.
+func GetObjectReaderWithOA(bp BaseParams, bck cmn.Bck, object string, options ...GetObjectInput) (r io.ReadCloser, oa *cmn.ObjAttrs, err error) {
+	var (
```

Comment:
I'll stop at this point. Take a look at CONTRIBUTING.md for general guidelines.

Again, lots of respect for the effort!